### ISSUE TITLE
Feature Request: Customizeable "Button"

### ISSUE THREAD

**Issue by JBlocklove** [mb1986/rm-hacks#61]
I'd love a way to set up a soft button which can execute user-configurable scripts. I have a program which automatically downloads the current day's NYT crossword, but currently it does it every day as soon as it gets an internet connection and that can be kind of annoying since it'll refresh xochitl and kick me out of the document I woke my remarkable up to read. It would be awesome if a button or set of buttons (configurable quantity maybe?) could be added to the taskbar which would execute a script or scripts determined by the user.

This is definitely a niche feature request, but I'd love to be able to take better advantage of the fact that the remarkable runs linux and users can write scripts to do any of a number of things (pull papers from arxiv, automatically sync documents with git, etc.).

I'm not the best software developer out there, but I'd be happy to help try to make this a reality if the team would be open to it.

**Comment by Eeems** [comment]
It might be worth just having places to load qml and letting users add whatever basic functionality they want.

**Comment by JBlocklove** [comment]
That would be awesome. More modularity in xochitl would be super useful since the hacks source code can't really be opened.

**Comment by mb1986** [comment]
I understand the need. I have to think about how to do that. The issue with the user provided qmls is that the user would need to know how to interact with other rM qml items (names, types).

**Comment by Eeems** [comment]
> I understand the need. I have to think about how to do that. The issue with the user provided qmls is that the user would need to know how to interact with other rM qml items (names, types).

Yes, but for them to build their own qml, you would expect them to already know this, so that shouldn't be an issue. I would recommend providing a couple examples for users to start with, and then the technical users would be able to create, share, and maintain their own.

**Comment by PepikVaio** [comment]
I join this request.
This would be great and expand the capabilities of reMarkable.
Good idea.
